index.js

```diff
@@ -54,6 +56,18 @@ function decorateLayer(layer, keysToKeep, newProps) {
         keepLookup[keys.indexOf(keysToKeep[i])] = true;
     }
 
+    if (requiredCount) {
+        for (i = 0; i < requiredCount; i++) {
+            requiredLookup[keys.indexOf(requiredKeys[i])] = true;
+        }
+
+        var features = layer.features.filter(function (feature) {
+            return hasAllKeys(feature.tags, requiredLookup, requiredCount);
+        });
+
+        layer.features = features;
```

Comment:
You can just do `layer.features = layer.features.filter(...)` since `features` array is not used after that.